dataactcore/interfaces/dbInterface.py

```diff
@@ -0,0 +1,40 @@
+import uuid
+
+from dataactcore.models.userModel import User, UserStatus
+from dataactcore.interfaces.db import databaseSession
+
+# First step to deprecating BaseInterface, its children, and corresponding
+# interface holders is to start moving all db access logic into one big
+# file (to prevent circular imports and have everything in the same place).
+# Still to do...make this work solely with a flask context...original idea
+# was that these functions would only be invoked within a Flask route, but
+# there are some (e.g., createUserWithPassword) that need to be here,
+# pending a further refactor.
+# As a temporary measure, until the next part of the work that refactors
+# the db access within Flask requests, fire up an ad-hoc db session in
+# these transitional functions.
+
+
+def createUserWithPassword(email, password, bcrypt, permission=1, cgac_code="SYS"):
+    """Convenience function to set up fully-baked user (used for setup/testing only)."""
+    with databaseSession() as sess:
+        status = sess.query(UserStatus).filter(UserStatus.name == 'approved').one()
+        user = User(email=email, user_status=status, permissions=permission,
+                    cgac_code=cgac_code)
+        pwd = getPasswordHash(password, bcrypt)
+        user.salt = pwd[0]
+        user.password_hash = pwd[1]
+        sess.add(user)
+        sess.commit()
+
+    return user
+
+
+def getPasswordHash(password, bcrypt):
+    """Generate password hash."""
+    # TODO: handle password hashing/lookup in the User model
+    salt = uuid.uuid4().hex
+    # number 12 below iw the number of rounds for bcrypt
+    hash = bcrypt.generate_password_hash(password + salt, 12)
```

Comment:
Yes, good call. I'll put it into a constant and will punt the config idea until we can get all of this password hashing stuff happening in the user model instead of here in the function bag.